src/Lykke.Exchange.Api.MarketData/Services/InitService.cs

```diff
@@ -18,81 +18,144 @@ public class InitService
         private readonly IDatabase _database;
         private readonly ILykkeMarketProfile _marketProfileClient;
         private readonly ICandleshistoryservice _candlesHistoryClient;
-        private readonly ITradesAdapterClient _tradesAdapterClient;
 
         public InitService(
             IDatabase database,
             ILykkeMarketProfile marketProfileClient,
-            ICandleshistoryservice candlesHistoryClient,
-            ITradesAdapterClient tradesAdapterClient
-            )
+            ICandleshistoryservice candlesHistoryClient
+        )
         {
             _database = database;
             _marketProfileClient = marketProfileClient;
             _candlesHistoryClient = candlesHistoryClient;
-            _tradesAdapterClient = tradesAdapterClient;
         }
 
         public async Task LoadAsync()
         {
-            List<MarketSlice> marketData = await GetMarketProfilesAsync();
-            Dictionary<string,IList<Candle>> todayCandles = await GetCandlesAsync();
-            
+            var now = DateTime.UtcNow;
+
+            var assetPairsTask = _candlesHistoryClient.GetAvailableAssetPairsAsync();
+            var marketDataTask = GetMarketProfilesAsync();
+
+            await Task.WhenAll(assetPairsTask, marketDataTask);
+
+            var marketData = marketDataTask.Result;
+
+            var todayCandlesTask = GetCandlesAsync(now.AddHours(-24), now.AddMinutes(5),
+                assetPairsTask.Result, CandlePriceType.Trades, CandleTimeInterval.Min5);
+            var lastMonthCandlesTask = GetCandlesAsync(now.AddYears(-1).AddHours(-24), now.AddMonths(1),
+                assetPairsTask.Result, CandlePriceType.Trades, CandleTimeInterval.Month);
+            var clearDataTask = ClearDataAsync(marketData.Select(x => x.AssetPairId).ToList());
+
+            await Task.WhenAll(todayCandlesTask, lastMonthCandlesTask, clearDataTask);
+
+
+            var todayCandles = todayCandlesTask.Result;
+            var lastMonthCandles = lastMonthCandlesTask.Result;
+
+
             foreach (var todayCandle in todayCandles)
             {
                 UpdateCandlesInfo(todayCandle.Key, todayCandle.Value, marketData);
             }
 
-            await UpdateLastPricesAsync(marketData);
+            foreach (var lastMonthCandle in lastMonthCandles)
+            {
+                UpdateLastPrice(lastMonthCandle.Key, lastMonthCandle.Value, marketData);
+            }
 
             await SaveMarketDataAsync(marketData, todayCandles);
         }
 
-        private async Task SaveMarketDataAsync(List<MarketSlice> marketData, Dictionary<string,IList<Candle>> prices)
+        private Task ClearDataAsync(List<string> assetPairIds)
+        {
+            var tasks = new List<Task>();
+
+            foreach (var assetPairId in assetPairIds)
+            {
+                tasks.Add(_database.KeyDeleteAsync(RedisService.GetMarketDataBaseVolumeKey(assetPairId)));
```

Comment:
There is a method for deleting an array of keys. The code would be more clean this way.